h2o-core/src/main/java/water/fvec/NewChunk.java

```diff
@@ -567,10 +567,25 @@ private void append_ss(BufferedString str) {
     byte b[] = str.getBuffer();
 
     if (_ss == null) {
-      _ss = MemoryManager.malloc1((strlen + 1) * 4);
+      int size = (strlen + 1) * 4;
+
+      if(size < 0 || size > Integer.MAX_VALUE - 9){
```

Comment:
Would be good to have a global constant for the MAX_VALUE - 9

Reply:
I've proposed a new class named H2OConstants with constants used throughout the project.